sway/config/output.c

```diff
@@ -314,11 +334,27 @@ static struct output_config *get_output_config(char *name, char *identifier) {
 	} else if (oc_name) {
 		// No identifier config, just return a copy of the name config
 		result = new_output_config(name);
+		if (config->reloading) {
+			default_output_config(result, sway_output->wlr_output);
+		}
 		merge_output_config(result, oc_name);
 	} else if (oc_id) {
 		// No name config, just return a copy of the identifier config
 		result = new_output_config(identifier);
+		if (config->reloading) {
+			default_output_config(result, sway_output->wlr_output);
+		}
 		merge_output_config(result, oc_id);
+	} else if (config->reloading) {
+		// Neither config exists, but we need to reset the output so create a
+		// default config for the output and if a wildcard config exists, merge
+		// that on top
+		result = new_output_config(identifier);
+		default_output_config(result, sway_output->wlr_output);
+		i = list_seq_find(config->output_configs, output_name_cmp, "*");
```

Comment:
Isn't this done already by `get_output_config`'s caller? Otherwise the wildcard config wouldn't be merged in case `oc_{name,id}` is not NULL.

Reply:
The caller will not have the wildcard config merged on top of the default output config. Without this outputs without their own output config will not get reset on reload and just have whatever is in the wildcard config itself changed.

Reply:
Hmm, I'm not sure why there's a need for a special case here.

There are three cases:

* There's no output config, no wildcard config: need to use the default one as is
* There's no output config, a wildcard config: need to use the default config with the wildcard config merged on top
* There's an output config: need to use the default config, merge the wildcard config on top if any, and _after_ that merge the output config on top

So I think `get_output_config` should always search for a wildcard config when reloading on line 311.

Reply:
Currently, the wildcard config gets merged on to all output configs when it is modified so the output config already has the information from the wildcard config. See `merge_wildcard_on_all` (L90) and `store_output_config` (L100).

Wildcard is currently treated as a special shorthand to apply to all rather than plain defaults. Changing it to apply wildcard here instead would be a breaking change. For example, if you manually set `dpms on` for a specific output, swayidle would not be able to change the dpms state with `output * dpms off` since the output specific `dpms on` would override it.

Reply:
Hmm, I see. Thanks for the explanation.